core/positions/market_position.go

```diff
@@ -71,6 +71,22 @@ func (p *MarketPosition) Closed() bool {
 	return p.size == 0 && p.buy+p.sell == 0
 }
 
+func (p *MarketPosition) UpdateInPlaceOnTrades(log *logging.Logger, traderSide types.Side, trades []*types.Trade, order *types.Order) *MarketPosition {
```

Comment:
In what context is that used, the naming suggest we are updating something, but it returns a clone, is that expected, could lead to some mistake if for the naming only in the future.